src/runtime/mod.rs

```diff
@@ -61,7 +62,10 @@ impl Runtime {
         let rt = tokio::runtime::Builder::new_current_thread()
```

Comment:
I think this is a good idea, but I'm trying to avoid touching too much existing code in this PR. I want to try and scope this refactor PR as narrowly as I can.